operations/src/test/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkServiceTest.java

```diff
@@ -0,0 +1,231 @@
+import java.util.zip.ZipFile;
+
+import static edu.unc.lib.boxc.model.fcrepo.test.TestHelper.makePid;
+import static edu.unc.lib.boxc.operations.impl.download.DownloadBulkService.getZipFilename;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class DownloadBulkServiceTest {
+    private static final String PARENT_UUID = "f277bb38-272c-471c-a28a-9887a1328a1f";
+    private static final String CHILD1_UUID = "83c2d7f8-2e6b-4f0b-ab7e-7397969c0682";
+    private static final String CHILD2_UUID = "0e33ad0b-7a16-4bfa-b833-6126c262d889";
+    private static final String FILENAME1 = "filename1.txt";
+    private static final String FILENAME2 = "filename2.txt";
+    private AutoCloseable closeable;
+    private PID parentPid;
+    private PID fileObject1Pid;
+    private PID fileObject2Pid;
+    private DownloadBulkService service;
+    private DownloadBulkRequest request;
+    @Mock
+    private AccessControlService aclService;
+    @Mock
+    private RepositoryObjectLoader repoObjLoader;
+    @Mock
+    private AccessGroupSet mockAccessSet;
+    @Mock
+    private AgentPrincipals mockAgent;
+    @Mock
+    private WorkObject parentWork;
+    @Mock
+    private FileObject fileObject1, fileObject2;
+    @TempDir
+    public Path zipStorageBasePath;
+
+    @BeforeEach
+    public void init() throws Exception {
+        closeable = openMocks(this);
+
+        service = new DownloadBulkService();
+        service.setAclService(aclService);
+        service.setRepoObjLoader(repoObjLoader);
+        service.setBasePath(zipStorageBasePath.toString());
+        parentPid = PIDs.get(PARENT_UUID);
+        fileObject1Pid = PIDs.get(CHILD1_UUID);
+        fileObject2Pid = PIDs.get(CHILD2_UUID);
+        request = new DownloadBulkRequest(PARENT_UUID, mockAgent);
+
+        when(mockAgent.getUsername()).thenReturn("user");
+        when(mockAgent.getPrincipals()).thenReturn(mockAccessSet);
+        when(fileObject1.getPid()).thenReturn(fileObject1Pid);
+        when(fileObject2.getPid()).thenReturn(fileObject2Pid);
+    }
+
+    @AfterEach
+    public void after() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    public void noAccessToWorkObjectTest() {
+        Assertions.assertThrows(AccessRestrictionException.class, () -> {
+            doThrow(new AccessRestrictionException()).when(aclService).assertHasAccess(
+                    anyString(), eq(parentPid), any(), eq(Permission.viewOriginal));
+
+            service.downloadBulk(request);
+        });
+    }
+
+    @Test
+    public void notAWorkTest() {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(null);
```

Comment:
when the requested object exists but is not a work, it'll actually throw a `ObjectTypeMismatchException` rather than returning null. I don't think the repositoryObjectLoader returns a null in other cases either, I think we get a `NotFoundException` if the object isn't found.